homeassistant/components/trafikverket_ferry/sensor.py

```diff
@@ -0,0 +1,143 @@
+        key="departure_time",
+        name="Departure Time",
+        icon="mdi:clock",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data["departure_time"],
+        info_fn=lambda data: data["departure_information"],
+    ),
+    TrafikverketSensorEntityDescription(
+        key="departure_from",
+        name="Departure From",
+        icon="mdi:ferry",
+        value_fn=lambda data: data["departure_from"],
+        info_fn=lambda data: data["departure_information"],
+    ),
+    TrafikverketSensorEntityDescription(
+        key="departure_to",
+        name="Departure To",
+        icon="mdi:ferry",
+        value_fn=lambda data: data["departure_to"],
+        info_fn=lambda data: data["departure_information"],
+    ),
+    TrafikverketSensorEntityDescription(
+        key="departure_modified",
+        name="Departure Modified",
+        icon="mdi:clock",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data["departure_modified"],
+        info_fn=lambda data: data["departure_information"],
+        entity_registry_enabled_default=False,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Trafikverket sensor entry."""
+
+    coordinator: TVDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            FerrySensor(coordinator, entry.data[CONF_NAME], entry.entry_id, description)
+            for description in SENSOR_TYPES
+        ]
+    )
+
+
+class FerrySensor(CoordinatorEntity[TVDataUpdateCoordinator], SensorEntity):
+    """Contains data about a ferry departure."""
+
+    entity_description: TrafikverketSensorEntityDescription
+    _attr_attribution = ATTRIBUTION
+
+    def __init__(
+        self,
+        coordinator: TVDataUpdateCoordinator,
+        name: str,
+        entry_id: str,
+        entity_description: TrafikverketSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._attr_name = f"{name} {entity_description.name}"
+        self._attr_unique_id = f"{entry_id}-{entity_description.key}"
+        self.entity_description = entity_description
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry_id)},
+            manufacturer="Trafikverket",
+            model="v1.2",
+            name=name,
+            configuration_url="https://api.trafikinfo.trafikverket.se/",
+        )
+        self._update_attr()
+
+    def _update_attr(self) -> None:
+        """Update _attr."""
+        self._attr_native_value = self.entity_description.value_fn(
+            self.coordinator.data
+        )
+        self._attr_extra_state_attributes = {
+            "other_information": self.entity_description.info_fn(self.coordinator.data),
```

Comment:
How often does this change?  

Entities that generate a significant amount of state changes can quickly increase the size of the database when the extra_state_attributes also change frequently. Minimize the number of extra_state_attributes for these entities by removing non-critical attributes or creating additional sensor entities.

Reply:
99.9% it comes back empty and doesn't change but for someone who is "subscribing" on their daily ferry home from work it can still be possible that something comes that way e.g. ferry cancelled or something else.
So I don't think it makes sense to have it as a sensor but it's worth to have it as an additional attribute.

This is definitively not something that will flood the DB.